components/post_view/channel_intro_message/channel_intro_message.jsx

```diff
@@ -555,23 +560,27 @@ function createInviteChannelButton(channel) {
     );
 }
 
-function createSetHeaderButton(channel) {
+function createSetHeaderButton(channel, formatMessage) {
     const channelIsArchived = channel.delete_at !== 0;
     if (channelIsArchived) {
         return null;
     }
+
+    const message = formatMessage({
```

Comment:
Instead of passing `formatMessage` all the way down here, you could use the version of `FormattedMessage` where you give is a function as a child
```
return (
    <FormattedMessage
        id='intro_messages.setHeader'
        defaultMessage='Set a Header'
    >
        {(message) => (
            <ToggleModalButtonRedux
                accessibilityLabel={message}
            >
                <EditIcon/>
                {message}
            </ToggleModalButtonRedux
        }}
    </FormattedMessage>
);
```

Reply:
@hmhealey changes made